ConsoleReader.java

```diff
@@ -17,13 +41,44 @@ public String askForMultiplayer() {
     return userInput;
   }
 
-  public String askForName(String name) {
-    System.out.printf("What will %s's name be?", name);
+  //A good optimization would be to make this method also only allow certain inputs. Then it could replace more functions.
+  //This would also fix the problem of not having any control for the animal input
+  //Used for any case in which the program asks for input and takes input without making certain the input is of a certain type
+  public String promptForStringInput(String prompt) {
+    System.out.printf(prompt);
     return(this.requester.nextLine());
   }
 
-  public String askForAnimalChoice(String chooserName) {
-    System.out.printf("%s, which animal would you like? Input (l) for lion, (e) for eagle, or (s) for shark");
-    return(requester.nextLine());
+  //Takes an animal and asks the user for all the AttributeValue values it needs.
+  public ArrayList<AttributeValue> askForAttributeValues(ArrayList<Attributes> requiredAnimalAttributes) {
+    System.out.printf("Please enter the values for the attributes of your animal. Each attribute must be given a value from 1-20, whole numbers only.");
+    String userInput;
+
+    //To hold the userInput after it becomes a number
+    int userInputValue;
+    ArrayList<AttributeValue> chosenAnimalAttributes = new ArrayList<AttributeValue>();
+    for (Attributes requiredAnimalAttribute : requiredAnimalAttributes) {
+      System.out.printf("What would you like the value of your %s attribute to be?", attributeToString.get(requiredAnimalAttribute));
+
+      //The loop will keep running until they give a integer from 1-20
+      while (true) {
+
+        //This checks that their input was an integer
+        userInput = this.requester.nextLine();
+        try {
+          userInputValue = Integer.parseInt(userInput);
+        } catch (NumberFormatException e) {
+          System.out.println("Please enter a whole number from 1-20");
```

Comment:
This message really apply to the scenario when the user enters a number outside of the [0,20] range?  If they don't enter a parsable number that's deserves a bit of different message, eh?